runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/time/Clock.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package aws.smithy.kotlin.runtime.time
+
+/**
+ * A source of time
+ */
+public interface Clock {
```

Comment:
**Comment**: My brazen re-appropriation of this class is [now pushed](https://github.com/awslabs/smithy-kotlin/commit/76f754d0950c5581a675d7804254acb976e1899a#diff-712258690a49644e675a8ef45306a56880884706c5bec294841d0e9a0e17db20). Feel free to rebase/merge or whatever.